s3migrate/README.md

```diff
@@ -0,0 +1,154 @@
+# Migrating from mogile to S3
+
+This folder contains code for migrating from Mogile to S3.
+
+In broad strokes, the migration from mogile to S3 consists of the following steps:
+
+1. Build and configure the migration.
+1. Sync data from mogile to an AWS bucket
+1. Remove or fix bad data.
+1. Run a large-scale migration in AWS lambda.
+1. Stop ingesting new data into contentrepo.
+1. Run a large-scale migration in AWS lambda.
+1. Verify migration.
+1. Update contentrepo configuration.
+1. Update data in contentrepo and rhino databases to point to the new, S3 buckets.
+1. Done!
+1. Test
+1. Re-enable ingest.
+
+
+## Details
+
+### Build and configure
+
+1. Create the production bucket, if it doesn’t exist.
+1. Set up access to the production bucket, including a key id and secret key for use by contentrepo.
+1. Create an SQS queue.
+1. Create a dynamodb table.
+1. Create a lambda function with full access to the SQS queue, the S3 buckets, and the dynamodb table created above.
```

Comment:
My take is that we should definitely manage the bucket the associated IAM user and Access Key in Pulumi, but I wonder about the rest of this setup (the SQS queue, Dynamodb table, Lambda function, and the associated roles and policies). Since that infrastructure is temporary, there's probably a strong case for hand-creating it in the console instead of investing in the creation of a pulumi stack for it. Though really, pulumi probably wouldn't be too onerous, and might help reduce some stress of accurately recreating this infra in prod, and tearing it down after it's no longer needed). What's your take?

Reply:
if it helps, here's what the pulumi looks like for counter https://github.com/PLOS/counter/pull/63/files#diff-f50c03398a19f9c7edfed291220501eb which has a somewhat similar architecture (but is probably a little more complicated)

Reply:
Yeah, I did this using AWS console and to me that seems like the "easiest" solution. To me that looks like a lot of pulumi code, but it's all new to me.

Reply:
Yeah, I'd estimate that it would take about 2 dev days to hammer out the pulumi code (which is a lot faster than Counter went, but I feel like understand pulumi much better), which doesn't seem worth it in terms of a raw time comparison, I think the main argument for pulumi would be to have the prod AWS infrastructure be predictable (and not have to deal with head scratching around missing policies and such).

I lean towards "no pulumi"

Reply:
OK, that kind of seems like an easy choice, since I think it should take about an hour to set up in console, max. It will immediately be obvious if there are issues with the setup, because the scripts and lambda function will fail with clear error messages.